packages/backend/src/backend_factory.ts

```diff
@@ -56,6 +60,16 @@ export class BackendFactory<
    * If no CDK App is specified a new one is created
    */
   constructor(constructFactories: T, stack: Stack = createDefaultStack()) {
+    if (
+      !AmplifyStorageFactory.hasDefault &&
+      AmplifyStorageFactory.factoryCounter > 1
+    ) {
+      throw new AmplifyUserError('NoDefaultBucketError', {
+        message: 'No default bucket set in the Amplify project.',
+        resolution:
+          'Add `isDefault: true` to one of the buckets `defineStorage` in the Amplify project.',
+      });
+    }
```

Comment:
Is there no other place for this validation than using static in Storage and using in the `backend_factory`? Perhaps using CDK.Lazy like here in function to delay this validation just before synth

https://github.com/aws-amplify/amplify-backend/blob/ec12ac83d730a37c539ff6ae2ee282341d4c5342/packages/backend-function/src/function_env_translator.ts#L43

Reply:
- [ ] TODO

Reply:
Deferring a calculation until before synths makes sense, but for validation, don't we want to fail fast?